plugins/snowflake/dbt/adapters/snowflake/connections.py

```diff
@@ -125,10 +126,28 @@ def _get_access_token(self) -> str:
             'Authorization': f'Basic {auth}',
             'Content-type': 'application/x-www-form-urlencoded;charset=utf-8'
         }
-        result = requests.post(token_url, headers=headers, data=data)
-        result_json = result.json()
+
+        result_is_json = False
+        result_json = {}
+        max_iter = 20
+        iter_count = 1
+        # Attempt to obtain JSON for 1 second before throwing an error
+        while not result_is_json:
+            if iter_count > max_iter:
+                break
+
+            result = requests.post(token_url, headers=headers, data=data)
+
+            if result.headers.get('content-type') == 'application/json':
```

Comment:
checking the content-type is totally reasonable, but I think a more robust approach would try/catch the `.json()` method call. That might look like (pseudo code):

```
result_json = None
for i in range(max_iter):
  result = requests.post(token_url, headers=headers, data=data)
  try:
    result_json = result.json()
    break
  except ValueError as e:
    message = response.text() # or something like that
    logger.debug(f"Got a non-json response ({result.status}): {e}, message: {message}")
    time.sleep(0.05)

if result_json is None:
  # We'd want to raise an error here, otherwise `result_json['access_token']` below will fail
  pass
```

What do you think?

Reply:
Right on. The only reason I recommended checking the content-type instead of try/except in my issue comment was from having read about the potential slowness of the exception-catching approach. On further reading/thinking, that should only matter for very large request responses, which this isn't.